bootstrap/013_task_queue.sql

```diff
@@ -0,0 +1,61 @@
+
+create or replace procedure task_queue.run_tasks()
+returns varchar
+language sql
+as
+begin
+show warehouses;
+show tasks in schema task_queue;
+let res resultset := (
+with tlist as (
+select "id" as id, "name" as name from table(result_scan(last_query_id())) where "name" <> 'run_tasks'
+),
+wlist as (
+select count(*) > 0 as wh_exists from table(result_scan(last_query_id(-2))) where "name" = 'SUNDECK_TASK_QUEUE_WH'
+),
+thist as (
+    select tlist.name, state, completed_time from tlist left outer join account_usage.task_history th on tlist.id = th.task_id where th.database_name = current_database() and schema_name='TASK_QUEUE'
+), tstate as (
+    select name, state, row_number() over (partition by name order by completed_time desc) as rn from thist
+), alltasks as (
+select 'task_queue.' || name as name, state from tstate where rn = 1)
+select case when state = 'SUCCEEDED' then 'drop task if exists ' || name || ';'
+when wh_exists then 'begin\nalter task '||name||' set warehouse = SUNDECK_TASK_QUEUE_WH;\nexecute task '|| name ||';\nend;'
+else 'begin\nalter task '||name||' unset warehouse;\nexecute task '|| name ||';\nend;'
+end as cmd from alltasks, wlist where state <> 'RUNNING'
+);
+let cur cursor for res;
```

Comment:
can you move to sql and just get back one block?

Reply:
fixed